project1/application.py

```diff
@@ -116,3 +120,47 @@ def userHome(Username):
     if Username in session:
         return render_template("userDetails.html", username=Username, message="Successfully logged in.", heading="Welcome back")
     return redirect(url_for('index'))
+
+
+
+@app.route("/search",methods=["POST","GET"])
+def search():
+	if request.method == "GET":
+		return render_template("search.html")
+	else:
+		result = request.form.get("search")
+		result = '%'+result+'%'
+		search_result = Books.query.filter(or_(Books.tittle.ilike(result), Books.author.ilike(result), Books.isbn.ilike(result),Books.year.ilike(result))).all()
+		return render_template("search.html", books=search_result)
+
+
+@app.route("/bookpage/<username>/<isbn>", methods = ["POST","GET"])
+def bookpage(username, isbn):
+	user1 = username
+
+	if user1 in session:
+		user_reviews = reviewRate.query.filter_by(isbn = book_isbn).all()
```

Comment:
How is this query relevant when you're making the same queries in both if and else part again

Reply:
removed query in else part